### DISCUSSION TITLE
xDrip+ & Libre2 direct (oop2) - sensor age issues prevent readings with latest sensor

### DISCUSSION THREAD

**Discussion by nigelkeirross** [NightscoutFoundation/xDrip#2777] [Q&A]
I am having problems with my current Libre2 sensor, reading direct (OOP2) with no bridge. Previous experience has been good - about 6 Libre 2 sensors direct - occasional disconnections, but usually reconnect by itself), but now I am only getting about 60% of the readings, (sometimes will go over an hour with no readings) often messages ""Sensor age has not advanced" or "sensor age has gone backwards". In system status the BT to the sensor is always connected (unless immediately following an NFC scan). These messages  also happen when I try NFC read from xDrip+ - only about 50% "work" even tho always "Scanned OK!" message. Are there any settings to override?

**Comment by PhilTU** [comment]
Do the missed readings also fail to appear in the OOP2 screen ? Is OOP2 up to date (last June or later)

**Reply by nigelkeirross** [reply]
@PhilTU They do seem to appear on the OOPS2 screen, and the version is from last June. Screenshot attached as well as latest xDrip+ screenshot. BTW, I have also had comments & help on this thread... https://github.com/NightscoutFoundation/xDrip/discussions/2760
![OOPS2-screen](https://user-images.githubusercontent.com/21008320/232484958-08d403ad-8650-4ec3-9eff-eb6c63aa86cf.png)
![latest-xdrip](https://user-images.githubusercontent.com/21008320/232485026-700942c5-f21a-44a0-8bef-5b7effa1ac72.png)

**Comment by PhilTU** [comment]
Interesting to see the Zero data values on the OOP2 screen, this has been raised elsewhere and from memory it was thought to be a sensor issue. Also the smoothing means that good values adjacent to zero values get deleted too.

**Reply by nigelkeirross** [reply]
@PhilTU Thanks. I am suspecting the sensor, but are you suggesting that I turn off "Smoothe Libre Data" in the less common / Other Misc settings? 
Also, from your other comment ""Was that Ooops or Ooops2? I already had both of those Libre2 settings turned on" - can you check this, I don't think settings related to Out of Process should be enabled for this configuration." Here are my relevant settings, I believe  for xdrip Display settings / Graph settings - trying to get raw data shown,  then Less Common/Advanced settings for Libre2 & finally Less Com,mon/Other Misc showing my OOP2 settings (calibration is set to be based on raw)...

![Settings-GraphSettings](https://user-images.githubusercontent.com/21008320/232503679-cf04c099-a936-4ecf-88ba-15e972825858.png)
![Settings-AdvancedForLibre2](h
![Settings-OtherMisc](https://user-images.githubusercontent.com/21008320/232503790-22325595-66b6-4345-9711-f0145d79bd16.png)
ttps://user-images.githubusercontent.com/21008320/232503743-cd6e20b9-20ae-444c-b184-6fe6231d3dbf.png)

**Reply by nigelkeirross** [reply]
![Settings-AdvancedForLibre2](https://user-images.githubusercontent.com/21008320/232504111-e2372839-8662-465e-b863-712b2b02182f.png)

**Comment by PhilTU** [comment]
![image](https://user-images.githubusercontent.com/76240903/232504875-3a1829bd-73c5-4422-bdde-2eb381296f91.png)
I was checking this was off, it is.

I agree about the sensor, if you look in the log you see values discarded where a sensible raw value was received but the smoothed value is zero or negative and hence the data point is rejected. So you get less data than you would expect. Turning off smoothing might help with that aspect, but the root cause is the zero values in the data stream.

**Comment by PhilTU** [comment]
Actually I may have been mistaken
![image](https://user-images.githubusercontent.com/76240903/232506304-9cdc59f7-d619-4220-9033-73068bc72aa1.png)
the glucoseLevelRaw is 1370 but the glucoseLevel=-1 on an example deleted data point, in this case from the NFC scan.

**Reply by nigelkeirross** [reply]
@PhilTU OK - thanks. I will try turning of Smoothe Libre Data & see what difference it makes.

**Reply by nigelkeirross** [reply]
@PhilTU This made no discernible difference. In the end I started a new sensor after 8 days with this one. Abbott are not interested in taking the faulty one back even for analysis as I was not using the approved app.
